### PR TITLE
KTOR-4078 update API docs for testing

### PR DESCRIPTION
**Pull request by andreyaksenov**

### CODE REVIEW

File: ktor-server/ktor-server-test-host/jvmAndNix/src/io/ktor/server/testing/TestApplication.kt [hunk 1]
```diff
@@ -193,7 +211,24 @@ public class ApplicationTestBuilder : TestApplicationBuilder(), ClientProvider {
...
+ * 1. Use [testApplication] function to set up a configured instance of a test application running locally.
+ * 2. Use the [HttpClient] instance inside a test application to make a request to your server,
+ * receive a response, and make assertions.
+ *
+ * The code below demonstrates how to test the most simple Ktor application that
+ * accepts GET requests made to the `/` path and responds with a plain text response:
+ * ```kotlin
+ * @Test
+ * fun testRoot() = testApplication {
+ *     val response = client.get("/")
```
Comment (rsinukov) [inline ktor-server/ktor-server-test-host/jvmAndNix/src/io/ktor/server/testing/TestApplication.kt:225]: Maybe add a route here?

File: ktor-server/ktor-server-test-host/jvmAndNix/src/io/ktor/server/testing/TestApplication.kt [hunk 2]
```diff
@@ -193,7 +211,24 @@ public class ApplicationTestBuilder : TestApplicationBuilder(), ClientProvider {
 }
 
 /**
- * Creates a test using [TestApplication]
+ * Creates a test using [TestApplication].
+ * To test a server Ktor application, do the following:
+ * 1. Use [testApplication] function to set up a configured instance of a test application running locally.
+ * 2. Use the [HttpClient] instance inside a test application to make a request to your server,
+ * receive a response, and make assertions.
+ *
```
Comment (rsinukov) [inline ktor-server/ktor-server-test-host/jvmAndNix/src/io/ktor/server/testing/TestApplication.kt:219]: I think that it's worth mentioning that this builder will apply config from `application.conf` file if it exists

**Review by rsinukov — APPROVED** [review]
LGTM, but minor comments

File: ktor-server/ktor-server-test-host/jvmAndNix/src/io/ktor/server/testing/TestApplication.kt (hunk 1 above)
Comment (andreyaksenov) [reply ktor-server/ktor-server-test-host/jvmAndNix/src/io/ktor/server/testing/TestApplication.kt:234]: thanks, fixed both comments

**Pull request merged**